core/src/main/java/org/sonarsource/sonarlint/core/container/connected/update/check/ModuleStorageUpdateChecker.java

```diff
@@ -0,0 +1,84 @@
+
+  private final StorageManager storageManager;
+  private final ModuleConfigurationDownloader moduleConfigurationDownloader;
+  private final GlobalPropertiesDownloader globalPropertiesDownloader;
+
+  public ModuleStorageUpdateChecker(StorageManager storageManager, ModuleConfigurationDownloader moduleConfigurationDownloader,
+    GlobalPropertiesDownloader globalPropertiesDownloader) {
+    this.storageManager = storageManager;
+    this.moduleConfigurationDownloader = moduleConfigurationDownloader;
+    this.globalPropertiesDownloader = globalPropertiesDownloader;
+  }
+
+  public StorageUpdateCheckResult checkForUpdates(String moduleKey, ProgressWrapper progress) {
+    DefaultStorageUpdateCheckResult result = new DefaultStorageUpdateCheckResult();
+    GlobalProperties globalProps = globalPropertiesDownloader.fetchGlobalProperties();
+
+    ModuleConfiguration serverModuleConfiguration = moduleConfigurationDownloader.fetchModuleConfiguration(moduleKey, globalProps);
+    ModuleConfiguration storageModuleConfiguration = storageManager.readModuleConfigFromStorage(moduleKey);
+
+    MapDifference<String, String> propDiff = Maps.difference(storageModuleConfiguration.getPropertiesMap(), serverModuleConfiguration.getPropertiesMap());
+    if (!propDiff.areEqual()) {
+      for (Map.Entry<String, String> entry : propDiff.entriesOnlyOnLeft().entrySet()) {
+        result.appendToChangelog(String.format("Property '%s' removed", entry.getKey()));
+      }
+      for (Map.Entry<String, String> entry : propDiff.entriesOnlyOnRight().entrySet()) {
+        result.appendToChangelog("Property '" + entry.getKey() + "' added with value '" + entry.getValue() + "'");
+      }
+      for (Map.Entry<String, ValueDifference<String>> entry : propDiff.entriesDiffering().entrySet()) {
+        result.appendToChangelog("Value of property '" + entry.getKey() + "' changed from '" + entry.getValue().leftValue() + "' to '" + entry.getValue().rightValue() + "'");
+      }
+    }
+
+    MapDifference<String, String> qProfileDiff = Maps.difference(storageModuleConfiguration.getQprofilePerLanguageMap(), serverModuleConfiguration.getQprofilePerLanguageMap());
+    if (!qProfileDiff.areEqual()) {
+      for (Map.Entry<String, String> entry : qProfileDiff.entriesOnlyOnLeft().entrySet()) {
+        result.appendToChangelog(String.format("Quality profile for language '%s' removed", entry.getKey()));
+      }
+      for (Map.Entry<String, String> entry : qProfileDiff.entriesOnlyOnRight().entrySet()) {
+        result.appendToChangelog("Quality profile for language '" + entry.getKey() + "' added with value '" + entry.getValue() + "'");
+      }
+      for (Map.Entry<String, ValueDifference<String>> entry : qProfileDiff.entriesDiffering().entrySet()) {
+        result.appendToChangelog(
+          "Quality profile for language '" + entry.getKey() + "' changed from '" + entry.getValue().leftValue() + "' to '" + entry.getValue().rightValue() + "'");
```

Comment:
I'm still skeptical about having the change log composed of strings.
To be seen later, perhaps we could have simply a boolean value flagging changes on different categories (properties, QPs, plugins, ..), or the number of changes. Categories would be enumerated.